chaoscenter/authentication/api/handlers/rest/project_handler.go

```diff
@@ -0,0 +1,679 @@
+package rest
+
+import (
+	"litmus/litmus-portal/authentication/api/presenter"
+	"litmus/litmus-portal/authentication/api/types"
+	"litmus/litmus-portal/authentication/pkg/entities"
+	"litmus/litmus-portal/authentication/pkg/services"
+	"litmus/litmus-portal/authentication/pkg/utils"
+	"litmus/litmus-portal/authentication/pkg/validations"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// GetUserWithProject returns user and project details based on username
+func GetUserWithProject(service services.ApplicationService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		username := c.Param("username")
+
+		// Validating logged in user
+		if c.MustGet("username").(string) != username {
+			log.Error("auth error: unauthorized")
+			c.JSON(utils.ErrorStatusCodes[utils.ErrUnauthorized],
+				presenter.CreateErrorResponse(utils.ErrUnauthorized))
+			return
+		}
+
+		user, err := service.FindUserByUsername(username)
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrUserNotFound], presenter.CreateErrorResponse(utils.ErrUserNotFound))
+			return
+		}
+
+		outputUser := user.GetUserWithProject()
+
+		projects, err := service.GetProjectsByUserID(outputUser.ID, false)
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
```

Comment:
Should we add error for "project not found" just like we have for "user not found" in line 37?

Reply:
@Jonsy13 the error here will not necessarily be project not found, it can be also some other error coming from the mongo side.